sdk/containerregistry/azure-containerregistry/azure/containerregistry/_exchange_client.py

```diff
@@ -0,0 +1,127 @@
+    SCOPE_PARAMETER = "scope"
+    SERVICE_PARAMETER = "service"
+    AUTHORIZATION = "Authorization"
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, TokenCredential, Dict[str, Any]) -> None
+        if not endpoint.startswith("https://"):
+            endpoint = "https://" + endpoint
+        self._endpoint = endpoint
+        self._credential_scopes = "https://management.core.windows.net/.default"
+        self._client = ContainerRegistry(
+            credential=credential,
+            url=endpoint,
+            sdk_moniker=USER_AGENT,
+            authentication_policy=ExchangeClientAuthenticationPolicy(),
+            credential_scopes=kwargs.pop("credential_scopes", self._credential_scopes),
+            **kwargs
+        )
+        self._credential = credential
+
+    def get_acr_access_token(self, challenge):
+        # type: (str) -> str
+        parsed_challenge = self._parse_challenge(challenge)
+        refresh_token = self.exchange_aad_token_for_refresh_token(service=parsed_challenge["service"])
+        return self.exchange_refresh_token_for_access_token(
+            refresh_token, service=parsed_challenge["service"], scope=parsed_challenge["scope"]
+        )
+
+    def exchange_aad_token_for_refresh_token(self, service=None):
+        # type: (str, Dict[str, Any]) -> str
+        refresh_token = self._client.authentication.exchange_aad_access_token_for_acr_refresh_token(
+            service=service, access_token=self._credential.get_token(self._credential_scopes).token
+        )
+        return refresh_token.refresh_token
+
+    def exchange_refresh_token_for_access_token(self, refresh_token, service=None, scope=None):
+        # type: (str, str, str) -> str
+        access_token = self._client.authentication.exchange_acr_refresh_token_for_acr_access_token(
+            service=service, scope=scope, refresh_token=refresh_token
+        )
+        return access_token.access_token
+
+    def _parse_challenge(self, header):
+        # type: (str) -> Dict[str, Any]
+        """Parse challenge header into service and scope"""
+        if header.startswith(self.BEARER):
+            challenge_params = header[len(self.BEARER) + 1 :]
+
+            matches = re.split(self.AUTHENTICATION_CHALLENGE_PARAMS_PATTERN, challenge_params)
+            self._clean(matches)
+            ret = {}
+            for i in range(0, len(matches), 2):
+                ret[matches[i]] = matches[i + 1]
+
+        return ret
+
+    def __enter__(self):
+        self._client.__enter__()
+        return self
+
+    def __exit__(self, *args):
+        self._client.__exit__(*args)
+
+    def close(self):
+        # type: () -> None
+        """Close sockets opened by the client.
+        Calling this method is unnecessary when using the client as a context manager.
+        """
+        self._client.close()
+
+    def _clean(self, matches):  # pylint: disable=no-self-use
+        # type: (List[str]) -> None
```

Comment:
Could we add a description of what this method is trying to achieve (maybe with input/output examples)?